### PR TITLE
doc: add filter section to endpoints that handle this feature

### PR DESCRIPTION
**Pull request by Tiity**

### CODE REVIEW

File: _includes/1.0/api/bankentry.html [hunk 1]
```diff
@@ -109,6 +109,48 @@ <h3 id="bankentryjsonformat">JSON Format</h3>
...
+            <td>status</td>
+            <td>
+                Filter by one or multiple status among :<br />
+                - UNPROCESSED<br />
+                - UNRECONCILED<br />
+                - PARTIALLY_RECONCILED<br />
+                - TO_REVIEW<br />
+                - RECONCILED<br />
+                - CLOSED.<br />
+                (e.g. ?status=partially_reconciled,reconciled).
```
Comment (fChalmeau) [inline _includes/1.0/api/bankentry.html:148]: Is it case sensitive ?

File: _includes/1.0/api/customer.html [hunk 2]
```diff
@@ -238,6 +238,32 @@ <h4>Individual only</h4>
...
 
+    <h3 id="customerfilter">Available Filter Fields</h3>
+    <table class="table table-bordered table-striped">
+        <thead>
+        {% include table-header-filter.html %}
+        </thead>
+        <tbody>
+        <tr>
+            <td>assigned_to</td>
+            <td>Filter by employee that is assigned to the customer.</td>
```
Comment (fChalmeau) [inline _includes/1.0/api/customer.html:249]: Can you add '...employee id that is...', to make it clear that the expected parameter is the id.

Maybe also add (e.g assigned_to=3) too.

File: _includes/1.0/api/customer.html [hunk 3]
```diff
@@ -238,6 +238,32 @@ <h4>Individual only</h4>
...
+        </tr>
+        <tr>
+            <td>created_date</td>
```
Comment (fChalmeau) [inline _includes/1.0/api/customer.html:252]: I think the param name is `created_at`.

**Review by fChalmeau — CHANGES_REQUESTED** [review]
Some improvments/changes to do.

I think we should also add a short 'Filter' section at the top of the documentation, that explains common rules and behavior of the filter parameters.

Like how we support and handle comparison operators ('AND', 'OR', GTE...), how you can mix filters togethers, use the same date param twice to filter within periods (like `created_at=gte:2020-10-01&created_at=lte:2020-10-31` to filter on October only)...

Also, in the same PR, can you turn the 'Sort' section into it's own section ? Like it's done for Pagination, Rate imit, Error codes, etc...

File: _includes/1.0/api/customer.html (hunk 3 above)
Comment (fChalmeau) [reply _includes/1.0/api/customer.html:252]: Same for the other `created_date` parameters of that PR.

File: _includes/1.0/api/bankentry.html (hunk 1 above)
Comment (Tiity) [reply _includes/1.0/api/bankentry.html:148]: Nop

Comment (fChalmeau) [reply _includes/1.0/api/bankentry.html:148]: Ok. Can you lower case them in the description ?

**Comment by Tiity** [comment]
@fChalmeau Fix with 73cd3a0

File: _includes/1.0/overview/filter.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<div id="filter" class="group">
+    <h2>Filter</h2>
+    <p>
+        API users are able to apply conditions on requests that support the filter feature.
+        Pass allowed parameters mentioned in each section named "Available Filter Options" of endpoints that handle this feature. 
+    </p>
+    <p>
+        Thanks to comparison and conjunction operators filter your allow to retrieve and shape data the way that you need it
```
Comment (fChalmeau) [inline _includes/1.0/overview/filter.html:8]: Thanks to comparison and conjunction operators filter, you're allowed to retrieve and shape data the way you need.
For instance, to retrieve realized accounting of a certain journal within 2020 first quarter, you can pass the following parameters to the "transations" end point :
?date=gte:2020-01-01%20AND%20lte:2020-03-31&journal_id=4&status=closed

File: _includes/1.0/overview/filter.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        {% include table-header-comparison-ops.html %}
+        </thead>
+        <tbody>
+        <tr>
+            <td>lte:</td>
+            <td><= (less than or equal)</td>
+        </tr>
+        <tr>
+            <td>gte:</td>
+            <td>>= (greater than or equal)</td>
```
Comment (fChalmeau) [inline _includes/1.0/overview/filter.html:24]: I guess we're missing operators here, like 'AND' and maybe others. Please test them on some end points, and if they are working properly then add them to the doc.

**Review by fChalmeau — CHANGES_REQUESTED** [review]
Needs to update some labels, and complete the Comparators list.

**Comment by Tiity** [comment]
> Needs to update some labels, and complete the Comparators list.

@fChalmeau fix with 0c1bbb5

**Comment by Tiity** [comment]
> Did you checked how it looks in the doc ?
> 
> I feel like we may need some BR here and there to make it look good.
> Also, can you use that style for the exemple ?
> 
> Capture d’écran 2020-11-13 à 18 36 54

@fChalmeau It looks like that
Screenshot 2020-11-17 at 12 02 58

**Review by fChalmeau — APPROVED** [review]
👍

**Pull request merged**